### PR TITLE
6 crud admin tasks

### PR DESCRIPTION
**Pull request by PyrinAndrii**
Screenshot 2022-04-12 at 22 47 06

Screenshot 2022-04-12 at 22 47 14

Screenshot 2022-04-12 at 22 48 40

### CODE REVIEW

File: app/models/repo/task.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    belongs_to :challenge
+    belongs_to :dependent_task, class_name: 'Repo::Task', optional: true
+
+    validates :title, presence: true
+    validates :title, uniqueness: { case_sensitive: true }
+    validates :slug, presence: true, if: :persisted?
+    validates :slug, uniqueness: true, if: :persisted?
+
+    validates :start_at, comparison: { less_than: :result_at }, allow_nil: true, if: :result_at
+    validates :submit_at, comparison: { less_than: :start_at }, allow_nil: true, if: :start_at
```
Comment (sveredyuk) [inline app/models/repo/task.rb]: submit_at should be greater than the start. Timeline is:
start doing -> submit work -> show results

File: db/migrate/20220411175218_create_tasks.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CreateTasks < ActiveRecord::Migration[7.0]
+  def change
+    create_table :tasks do |t|
+      t.citext :title, null: false, index: { unique: true }
```
Comment (sveredyuk) [inline db/migrate/20220411175218_create_tasks.rb:4]: unique in scope of the challenge

File: db/migrate/20220411175218_create_tasks.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      t.citext :slug, null: false, index: { unique: true }
+      t.text :description
+      t.references :challenge, null: false, foreign_key: true
```
Comment (sveredyuk) [inline db/migrate/20220411175218_create_tasks.rb:7]: + index

**Review by sveredyuk — COMMENTED** [review]
Good job.
+ Specs and merge it

File: app/models/repo/task.rb (hunk 1 above)
Comment (PyrinAndrii) [reply app/models/repo/task.rb]: added it here [update task validations](https://github.com/DevChallengeIT/devchallenge-platform/pull/19/commits/81e85721125592167133b5b939a6138f74cd7af0)

File: db/migrate/20220411175218_create_tasks.rb (hunk 2 above)
Comment (PyrinAndrii) [reply db/migrate/20220411175218_create_tasks.rb:4]: Added validation to model here - [update task validations](https://github.com/DevChallengeIT/devchallenge-platform/pull/19/commits/81e85721125592167133b5b939a6138f74cd7af0)

Is it what you meant?

File: db/migrate/20220411175218_create_tasks.rb (hunk 3 above)
Comment (PyrinAndrii) [reply db/migrate/20220411175218_create_tasks.rb:7]: we have the index, it was added automatically. See here line 50 - https://github.com/DevChallengeIT/devchallenge-platform/blob/81e85721125592167133b5b939a6138f74cd7af0/db/schema.rb#L38-L54

**Comment by PyrinAndrii** [comment]
@sveredyuk, I pushed changes.

I will add tests in the evening.
But if the PR blocks us now, I think we can merge it, and add tests for tasks in a separate PR. What do you think?

**Comment by PyrinAndrii** [comment]
> Good job.
> 
> * Specs and merge it

Added specs

**Review by sveredyuk — APPROVED** [review]
Good job

**Pull request merged**